src/signs/ui/search/Signs.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import { parse } from 'query-string'
+
+import AppContent from 'common/AppContent'
+import SessionContext from 'auth/SessionContext'
+
+import { SectionCrumb } from 'common/Breadcrumbs'
+import { Session } from 'auth/Session'
+import SignsSearchForm from 'signs/ui/search/SignsSearchForm'
+import SignsSearch from 'signs/ui/search/SignsSearch'
+import { SignQuery } from 'signs/domain/Sign'
+import _ from 'lodash'
+
+function parseQuery(value: string | null | string[]): string | undefined {
+  if (typeof value === 'string' && value !== '') {
+    return value
+  } else {
+    return undefined
```

Comment:
Do not return `undefined`, use `null`.

Reply:
So i can return null but i would have to convert it to undefined later on `sign.repository.search(stringify(signQuery))`. So the api decides on what to do depending on which query parameter are set and the rest has to be undefined.
Example currently api request: `signs?sign=ba&subIndex=1&value=ba`
This would break the api if using null because the api request would be : `signs?isComposite&isIncludeHomophones&sign=ba&subIndex=1&value=ba`

Reply:
You need to fix your request. Do not make code that depends on JS quirks. Hint: see [query-string](https://github.com/sindresorhus/query-string) documentation.

`undefined` is used when something is not defined. Given that defining something as `undefined` does not make sense and can make the code harder to understand. If you need to get rid of some properties you can filter with lodash.